lib/actions/deploy.coffee

```diff
@@ -225,16 +225,21 @@ module.exports =
 			if image? and options.build
 				throw new ExpectedError('Build option is not applicable when specifying an image')
 		.then ->
-			validateProjectDirectory(sdk, {
-				dockerfilePath: options.dockerfile,
-				noParentCheck: options['noparent-check'] || false,
-				projectPath: options.source || '.',
-				registrySecretsPath: options['registry-secrets'],
-			})
-		.then ({ dockerfilePath, registrySecrets }) ->
-			options.dockerfile = dockerfilePath
-			options['registry-secrets'] = registrySecrets
-
+			if image
+				getRegistrySecrets(sdk, options['registry-secrets'])
+				.then (registrySecrets) ->
+					options['registry-secrets'] = registrySecrets
+			else
+				validateProjectDirectory(sdk, {
+					dockerfilePath: options.dockerfile,
+					noParentCheck: options['noparent-check'] || false,
+					projectPath: options.source || '.',
+					registrySecretsPath: options['registry-secrets'],
+				})
+				.then ({ dockerfilePath, registrySecrets }) ->
+					options.dockerfile = dockerfilePath
+					options['registry-secrets'] = registrySecrets
+		.then ->
```

Comment:
The explanation is that when I added project directory validation in #1571, I forgot that the `balena deploy <image>` command, for a pre-built image, does not use any project directory at all (nothing to build), so the current working directory should not be validated as if it was a project directory.